astropy/visualization/lupton_rgb.py

```diff
@@ -0,0 +1,477 @@
+
+
+class AsinhZScaleMapping(AsinhMapping):
+    """
+    A mapping for an asinh stretch, estimating the linear stretch by zscale.
+
+    x = asinh(Q (I - z1)/(z2 - z1))/Q
+
+    See AsinhMapping
+
+    """
+
+    def __init__(self, image1, image2=None, image3=None, Q=8, pedestal=None):
+        """
+        Create an asinh mapping from an image, setting the linear part of the
+        stretch using zscale.
+
+        Parameters
+        ----------
+        image1 : `~numpy.ndarray` or a list of arrays
+            The image to analyse, or a list of 3 images to be converted to
+            an intensity image.
+        image2 : `~numpy.ndarray`, optional
+            the second image to analyse (must be specified with image3).
+        image3 : `~numpy.ndarray`, optional
+            the third image to analyse (must be specified with image2).
+        Q : float, optional
+            The asinh softening parameter. Default is 8.
+        pedestal : float or sequence(3), optional
+            The value, or array of 3 values, to subtract from the images; or None.
+
+        Notes
+        -----
+        N.b. pedestal, if not None, is removed from the images when
+        calculating the zscale stretch, and added back into
+        Mapping.minimum[]
+        """
+
+        if image2 is None or image3 is None:
+            if not (image2 is None and image3 is None):
+                raise ValueError("please specify either a single image "
+                                 "or three images.")
+            image = [image1]
+        else:
+            image = [image1, image2, image3]
+
+        if pedestal is not None:
+            try:
+                len(pedestal)
+            except TypeError:
+                pedestal = 3*[pedestal]
+
+            if len(pedestal) != 3:
+                raise ValueError("please provide 1 or 3 pedestals.")
+
+            image = list(image)        # needs to be mutable
+            for i, im in enumerate(image):
+                if pedestal[i] != 0.0:
+                    image[i] = im - pedestal[i]  # n.b. a copy
+        else:
+            pedestal = len(image)*[0.0]
+
+        image = compute_intensity(*image)
+
+        zscale_limits = ZScaleInterval().get_limits(image)
+        zscale = LinearMapping(*zscale_limits, image=image)
+        data_range = zscale.maximum - zscale.minimum[0]  # zscale.minimum is always a triple
+        minimum = zscale.minimum
+
+        for i, level in enumerate(pedestal):
+            minimum[i] += level
+
+        AsinhMapping.__init__(self, minimum, data_range, Q)
+        self._image = image
+
+
+def make_lupton_rgb(image_r, image_g=None, image_b=None, minimum=0, data_range=5, Q=8,
+                    saturated_border_width=0, saturated_pixel_value=None,
+                    x_size=None, y_size=None, rescale=None,
+                    filename=None):
+    """
+    Make an RGB color image from 3 images using an asinh stretch.
+
```

Comment:
Can you please link to the high-level docs section about this function from the docstring?

Here's an example of what I mean: a class `Galactocentric` which has a separate page in `docs` and the class docstring and separate page are cross-linked:
http://astropy.readthedocs.io/en/latest/coordinates/galactocentric.html